packages/odyssey-react-mui/src/labs/SideNav/SideNav.tsx

```diff
@@ -450,11 +480,16 @@ const SideNav = ({
                 );
               } else {
                 return (
-                  <SideNavItemContent
+                  <SideNavItemContentContext.Provider
+                    value={{ isCompact }}
                     key={item.id}
-                    scrollRef={getRefIfThisIsFirstNodeWithIsSelected(item.id)}
-                    {...item}
-                  />
+                  >
+                    <SideNavItemContent
+                      key={item.id}
+                      scrollRef={getRefIfThisIsFirstNodeWithIsSelected(item.id)}
```

Comment:
I feel like there's a better way of doing this. And because I know function refs are deprecated in React v19, I'd prefer we find a better way of doing it. But it would probably require 2 refs.